### ISSUE TITLE
Trackmania² Stadium, Canyon, Lagoon & Valley (232910 228760 600720 243360)

### ISSUE THREAD

**Issue by nyannoying1337** [ValveSoftware/Proton#193]
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i3-6100 CPU @ 3.70GHz
    CPU Family:  0x6
    CPU Model:  0x5e
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  3700 Mhz
    4 logical processors
    2 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Ubuntu 18.04.1 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.15.0-32-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11906000
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime-beta-release_2018-06-14

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1050 Ti/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 396.51
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x10de
    DeviceID:  0x1c82
    Revision Not Detected
    Number of Monitors:  2
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 3840 x 1080
    Primary Display Size: 21.42" x 11.93" (24.49" diag)
                                            54.4cm x 30.3cm (62.2cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 4096 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC887-VD

Memory:
    RAM:  7902 Mb

Miscellaneous:
    UI Language:  English
    LANG:  de_DE.UTF-8
    Total Hard Disk Space Available:  18659 Mb
    Largest Free Hard Disk Block:  8793 Mb
    VR Headset: None detected

Recent Failure Reports:

**Comment by thunder1410** [comment]
Can confirm that trackmania2 stadium works without any issues.
system information: https://gist.github.com/thunder1410/76897c9bd211c74b30968baa4037ce96

**Comment by cverstege** [comment]
I can confirm too all those titles are working out of the box.
System information: https://gist.github.com/Citiroller/9451e491fd43078bf3871b56b83a4cf9

**Comment by cverstege** [comment]
Shootmania Storm works fine too, see issue #776

**Comment by cverstege** [comment]
When those games are added to the Whitelist, they should use the same wine prefix, including Shootmania Storm (229870).
The steam install installs only the "Engine" (Maniaplanet) while the game Data is actually then downloaded from the game to the local Documents folder.
On windows it's therefore possible to switch between the different Trackmania² titles within the same process of "Maniaplanet", you don't have to quit the game and start a different title.
For Example:
On Windows I can launch TM Canyon and switch to TM Stadium without the game restarting. Steam still thinks I'm playing Canyon as it's still the same process of Maniaplanet which can handle all other titles.
With proton I can do the same, but it would download the Stadium packs again in my Canyon prefix, as it's only installed on the Stadium prefix for example.
Also the saves (replays, custom maps, etc) aren't synced between all titles, because of the different prefixes.
When I'm now starting the Stadium Title from within the Canyon Prefix (which is possible due to the game/engine (Maniaplanet) design as explained in the beginning) i don't see the replays of my best times i drove in the Stadium prefix.

I hope I could explain this issue good enough for those who don't know how all the Maniaplanet titles interact with each other. All in all, all the Trackmania² Titles and Shootmania Storm should use the same wine prefix to share the same user and documents folder.

**Comment by DarkArc** [comment]
~~Trackmania2 Stadium isn't launching at all for me.~~

~~Sys info: https://gist.github.com/DarkArc/050974b1e3f836b8877d5452893d1ad5~~

Edit: Please disregard, system is using out of date drivers.

**Comment by cverstege** [comment]
Maybe your graphics drivers are too old. Accordinf to the Proton documentation you need Mesa 18.1.x for DX11 games at minimum. you have Mesa 18.0.5 according to your System Information.
The Maniaplane Titles require DX10 or 11.
Edit: Some stores say it requires DX11, so update Mesa and it should work?

**Comment by DarkArc** [comment]
Good catch @Citiroller, I'm not currently able to jump to 18.04. When I'm on 18.04 w/ the graphics PPA Valve mentions, I'll come back and update my report. Looks like @thunder1410 has virtually the same graphics card I do.

**Comment by thunder1410** [comment]
@DarkArc Make sure python 2.7 is installed or you are using the latest version of proton. The default 3.7-3 proton only has support for python 2.7 not python 3.x.

**Comment by mimattr** [comment]
**Trackmania² Stadium (232910) & Trackmania² Lagoon (600720)**

Ubuntu 16.04, NVIDIA GTX 970, Proton 3.7-5 Beta:

[Steam System Information](https://gist.github.com/mimattr/b0b7f46366e31e254460b9593ce87e51)

Test | Result
-- | --
Singleplayer | Working as expected
Local Co-op | _NOT TESTED^_
Online Multiplayer | Working as expected
VR | _NOT TESTED_

Configuration | ...
-- | --
Input | Steam Controller
Display | 1920x1080
Fullscreen | Yes
Preset | _N/A_
VSync | _N/A_
API | DX11

^ Currently not supported in [Maniaplanet 4](https://www.maniaplanet.com) launcher

_NOTE: **Maniaplanet** common files are currently duplicated using `$WINEPREFIX`, recommend sharing `%ProgramData%\Maniaplanet` & `%USERPROFILE%\My Documents\Maniaplanet` across all related products._

**Comment by DarkArc** [comment]
Working great for me after Proton 3.16-4 Beta :)

**Comment by kisak-valve** [comment]
Stuttering or Unrecognized/Incorrectly Mapped Controller in TrackMania² Stadium

Issue transferred from https://github.com/ValveSoftware/Proton/issues/3661.
@AuroransSolis posted on 2020-03-20T08:56:43:

# Compatibility Report
- Name of the game with compatibility issues: TrackMania² Stadium
- Steam AppID of the game: 232910

## System Information
- GPU: AMD Radeon RX 480
- Driver/LLVM version: Mesa 20.0.1
- Kernel version: 5.4.2-gentoo
- Link to full system information report as [Gist](https://gist.github.com/AuroransSolis/eae682d0e5c5a3514a525cf4f49a336f):
- Proton versions: 5.0-4, 5.0-3, 4.11-13, 4.2-9, 3.16-9

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

Log using Proton 3.16-9: [steam-232910-3.16-9.log](https://github.com/ValveSoftware/Proton/files/4358543/steam-232910-3.16-9.log)
Log using Proton 4.2-9: [steam-232910-4.2-9.log](https://github.com/ValveSoftware/Proton/files/4358539/steam-232910-4.2-9.log)
Log using Proton 4.11-13: [steam-232910-4.11-13.log](https://github.com/ValveSoftware/Proton/files/4358540/steam-232910-4.11-13.log)
Log using Proton 5.0-3: [steam-232910-5.0-3-1.log](https://github.com/ValveSoftware/Proton/files/4358590/steam-232910-5.0-3-1.log)
Log using Proton 5.0-3: [steam-232910-5.0-3-2.log](https://github.com/ValveSoftware/Proton/files/4358592/steam-232910-5.0-3-2.log)
Log using Proton 5.0-4: [steam-232910-5.0-4-1.log](https://github.com/ValveSoftware/Proton/files/4358541/steam-232910-5.0-4-1.log)
Log using Proton 5.0-4: [steam-232910-5.0-4-2.log](https://github.com/ValveSoftware/Proton/files/4358542/steam-232910-5.0-4-2.log)

Note: there's a difference in the settings and behaviour of the logs using 5.0-{3,4}, and I believe they are related. In the first one for each, the game is being run with the "Steam Input Per-Game Setting" set to "Use your global Steam Input settings (Xbox)". This was unintentional for the very first log I made (5.0-4), but I changed it to "Forced off" when I realized that the controller wasn't recognized in the game. Curiously, when it wasn't set to "Forced off", the game ran without stuttering, but as soon as I changed it back to that it started stuttering again. I don't know exactly what's going on here, but I do find it interesting.

## Symptoms
I have two choices with the game presently:
1. Extreme stuttering that makes the game unplayable (5.0-4, 5.0-3, 4.11-13)
2. Unrecognized/incorrectly mapped controller (4.2-9, 3.16-9)

## Reproduction
1. Choose any of the mentioned Proton versions
    a) Set "Steam Input Per-Game Setting" to "Forced off" or "Use your global Steam Input settings (Xbox)" for your choice of unplayable stuttering or unrecognized controller respectively if using >= 4.11-13
2. Start the game

**Comment by k4rli** [comment]
TrackMania2 Canyon
nvidia-driver-510, Ubuntu 21.10
Works only with 5.0-10. Experimental, 7.0-1, 6.3-8, 5.13-6 crash on startup.
